src/Chainweb/Pact/PactService.hs

```diff
@@ -1039,24 +1043,28 @@ applyPactCmd
     -> ChainwebTransaction
     -> Miner
     -> ModuleCache
-    -> ([HashCommandResult] -> [HashCommandResult])  -- ^ difference list
-    -> PactServiceM cas (T2 ([HashCommandResult] -> [HashCommandResult]) ModuleCache)
+    -> DList HashCommandResult
+    -> PactServiceM cas (T2 (DList HashCommandResult) ModuleCache)
 applyPactCmd isGenesis dbEnv cmdIn miner mcache dl = do
     psEnv <- ask
     let !logger   = _cpeLogger . _psCheckpointEnv $ psEnv
         !pd       = _psPublicData psEnv
         !spv      = _psSpvSupport psEnv
         pactHash  = view P.cmdHash cmdIn
 
-    T2 !result mcache' <- liftIO $ if isGenesis
+    T2 result mcache' <- liftIO $ if isGenesis
         then applyGenesisCmd logger dbEnv pd spv (payloadObj <$> cmdIn)
         else applyCmd logger dbEnv miner (_psGasModel psEnv) pd spv cmdIn mcache
 
+    -- on genesis, seed initial service state with loaded modules
+    when isGenesis $
+      psInitCache .= mcache'
```

Comment:
When we're at genesis, seed the cache.

Reply:
Will this still work for nodes that are way past genesis height? Does this trigger upon startup?

Reply:
This is something I'd want to talk with stuart about. I don't know. Tbh i think this would only work in general for services starting at 0. I'm not honestly sure how to get older pact services synced up otherwise.